2019/d01p1FuelCounterUpper.d

```diff
@@ -1,14 +1,23 @@
 import std.algorithm;
 
+int fuelFor(int mass) {
+  int fuel = mass / 3 - 2;
+  return fuel <= 0 ? 0 : fuel + fuelFor(fuel);
```

Comment:
Why recursion here?

Reply:
`fuel + fuelFor(fuel)` seems the most natural way to express this. Besides, the compiler knows what I mean and will produce the best code with tail call optimisation.

Reply:
Point taken on expressiveness but how do you know the compiler is producing the best code?

Reply:
The awesome and majestic [Compiler Explorer](https://gcc.godbolt.org) pronounced "god bolt". I've prepared some samples.

Compiled with [LDC 1.18.0 unoptimised](https://gcc.godbolt.org/z/_L2rHM) clearly generates recursive code.

Compiled with [LDC 1.18.0 optimised](https://gcc.godbolt.org/z/WjLoEN) frankly I don't know what the fuck it does, but it looks faster.